common/djangoapps/entitlements/models.py

```diff
@@ -336,45 +347,59 @@ def get_fullfillable_entitlement_for_user(cls, user, course_run_key):
             CourseEntitlement: The most recent fullfillale CourseEntitlement, None otherwise.
         """
         # Check if the User has any Entitlements that are active
-        if cls.does_user_have_active_entitlements(user):
+        entitlements = cls.get_fullfillable_entitlements(user)
+        if entitlements:
             course_uuid = get_course_uuid_for_course(course_run_key)
             if course_uuid:
-                # Look for an entitlement for the current uuid
-                try:
-                    course_entitlement = cls.objects.filter(
-                        user=user,
-                        course_uuid=course_uuid
-                    ).exclude(
-                        expired_at__isnull=False,
-                        enrollment_course_run__isnull=False
-                    ).latest('created')
-                except CourseEntitlement.DoesNotExist:
-                    return None
-
-                # Verify that the enrollment can be enrolled in
-                if not is_course_run_entitlement_fullfillable(
-                        course_run_key=course_run_key,
-                        entitlement=course_entitlement
-                ):
-                    return None
-                return course_entitlement
+                for entitlement in entitlements:
```

Comment:
rather than iterating through all of the user's fulfillable entitlements, couldn't the set be filtered be course_uuid in line 350?

Reply:
No we do not have the course_uuid at that point.  This was done to avoid calling the API for the course_uuid when the user does not have any entitlements. 

I will add a comment.